### PR TITLE
Rename `csympy` --> `symengine`

### PR DESCRIPTION
**Pull request by thilinarmtb**

### CODE REVIEW

File: symengine/lib/symengine_wrapper.pyx [hunk 1]
```diff
@@ -9,50 +9,50 @@ from cpython cimport PyObject, Py_XINCREF, Py_XDECREF, \
...
         r = FunctionWrapper.__new__(FunctionWrapper)
-    elif (csympy.is_a_RealDouble(deref(o))):
+    elif (symengine.is_a_RealDouble(deref(o))):
         r = RealDouble.__new__(RealDouble)
-    elif (csympy.is_a_ComplexDouble(deref(o))):
+    elif (symengine.is_a_ComplexDouble(deref(o))):
         r = ComplexDouble.__new__(ComplexDouble)
     else:
-        raise Exception("Unsupported CSymPy class.")
+        raise Exception("Unsupported symengine class.")
```
Comment (thilinarmtb) [inline symengine/lib/symengine_wrapper.pyx]: This should be `SymEngine`.

File: symengine/lib/symengine_wrapper.pyx [hunk 2]
```diff
@@ -9,50 +9,50 @@ from cpython cimport PyObject, Py_XINCREF, Py_XDECREF, \
...
     r.thisptr = o
     return r
 
-def sympy2csympy(a, raise_error=False):
+def sympy2symengine(a, raise_error=False):
     """
-    Converts 'a' from SymPy to CSymPy.
+    Converts 'a' from SymPy to symengine.
```
Comment (thilinarmtb) [inline symengine/lib/symengine_wrapper.pyx]: `SymEngine`

**Comment by certik** [comment]
I think we initially wanted to use csympy for the Python wrappers, but I am fine either way.

File: symengine/tests/test_sympy_conv.py [hunk 3]
```diff
@@ -1,14 +1,14 @@
-from csympy import (Symbol, Integer, sympify, SympifyError, sin, cos,
+from symengine import (Symbol, Integer, sympify, SympifyError, sin, cos,
         function_symbol, I, E, pi, exp)
-from csympy.lib.csympy_wrapper import densematrix, Subs, Derivative
+from symengine.lib.symengine_wrapper import densematrix, Subs, Derivative
 import sympy
 
-# Note: We test _sympy_() for CSymPy -> SymPy conversion, as those are methods
+# Note: We test _sympy_() for symengine -> SymPy conversion, as those are methods
```
Comment (thilinarmtb) [inline symengine/tests/test_sympy_conv.py]: `SymEngine`

File: symengine/tests/test_sympy_conv.py [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 # that are implemented in this library. Users can simply use sympy.sympify(...)
 # to do this conversion, as this function will call our _sympy_() methods under
 # the hood.
 #
-# For SymPy -> CSymPy, we test csympy.sympify(...) which does the conversion.
+# For SymPy -> symengine, we test symengine.sympify(...) which does the conversion.
```
Comment (thilinarmtb) [inline symengine/tests/test_sympy_conv.py]: `SymEngine`

**Comment by thilinarmtb** [comment]
Ah, I didn't know that. That makes sense.
@isuruf : WDYT?

**Comment by certik** [comment]
I am fine with naming it symengine. It will be consistent with Julia and Ruby wrappers.

**Comment by Sumith1896** [comment]
Isn't there possibility of confusion with plain `SymEngine` for wrappers?
Since it resides in the same repo.

**Comment by thilinarmtb** [comment]
@certik : Great !! I'll fix the errors.

@Sumith1896 : It was called `csympy` when the repo was `csympy`?

**Comment by Sumith1896** [comment]
Oh It was completely out of mind that previously we called it `csympy`. I think it is fine then.

**Comment by certik** [comment]
There are also test failures like:
```
+mkdir -p empty
+cd empty
+cat
+python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named csympy
```
So you need to update the Travis tests.

I am fine with naming the python module symengine. I think it's a better name and less confusion.

**Comment by Sumith1896** [comment]
+1

**Comment by certik** [comment]
This is a good change, I think it's the way to go. I am merging it.

**Comment by thilinarmtb** [comment]
Thanks for merging !!

**Pull request merged**